src/transactions/test/PathPaymentStrictSendTests.cpp

```diff
@@ -509,65 +509,86 @@ TEST_CASE("pathpayment strict send", "[tx][pathpayment]")
+    SECTION("issuer missing")
     {
-        auto market = TestMarket{*app};
-        auto source = root.create("source", minBalance1);
-        auto destination = root.create("destination", minBalance1);
-        source.changeTrust(idr, 20);
-        destination.changeTrust(usd, 20);
-        gateway.pay(source, idr, 10);
-        for_versions_from(12, *app, [&] {
-            gateway.merge(root);
-            REQUIRE_THROWS_AS(source.pathPaymentStrictSend(destination, idr, 10,
-                                                           usd, 10, {}, &idr),
-                              ex_PATH_PAYMENT_STRICT_SEND_NO_ISSUER);
-            // clang-format off
-            market.requireBalances(
-                {{source, {{xlm, minBalance1 - 2 * txfee}, {idr, 10}, {usd, 0}}},
-                 {destination, {{xlm, minBalance1 - txfee}, {idr, 0}, {usd, 0}}}});
-            // clang-format on
-        });
-    }
+        // look at the SECTION "uses all offers in a loop" for a
+        // successful path payment test with no issuers
 
-    SECTION("middle issuer missing")
-    {
         auto market = TestMarket{*app};
         auto source = root.create("source", minBalance1);
         auto destination = root.create("destination", minBalance1);
         source.changeTrust(idr, 20);
         destination.changeTrust(usd, 20);
         gateway.pay(source, idr, 10);
-        for_versions_from(12, *app, [&] {
-            auto btc = makeAsset(getAccount("missing"), "BTC");
-            REQUIRE_THROWS_AS(source.pathPaymentStrictSend(
-                                  destination, idr, 10, usd, 10, {btc}, &btc),
-                              ex_PATH_PAYMENT_STRICT_SEND_NO_ISSUER);
-            // clang-format off
-            market.requireBalances(
-                {{source, {{xlm, minBalance1 - 2 * txfee}, {idr, 10}, {usd, 0}}},
-                 {destination, {{xlm, minBalance1 - txfee}, {idr, 0}, {usd, 0}}}});
-            // clang-format on
-        });
-    }
 
-    SECTION("last issuer missing")
-    {
-        auto market = TestMarket{*app};
-        auto source = root.create("source", minBalance1);
-        auto destination = root.create("destination", minBalance1);
-        source.changeTrust(idr, 20);
-        destination.changeTrust(usd, 20);
-        gateway.pay(source, idr, 10);
         for_versions_from(12, *app, [&] {
-            gateway2.merge(root);
-            REQUIRE_THROWS_AS(source.pathPaymentStrictSend(destination, idr, 10,
-                                                           usd, 10, {}, &usd),
-                              ex_PATH_PAYMENT_STRICT_SEND_NO_ISSUER);
+            uint32_t ledgerVersion;
+            {
+                LedgerTxn ltx(app->getLedgerTxnRoot());
+                ledgerVersion = ltx.loadHeader().current().ledgerVersion;
+            }
+
+            SECTION("path payment send issuer missing")
+            {
+                gateway.merge(root);
+                if (ledgerVersion < 13)
```

Comment:
Nitpick: this "if-else" appears 3x. Can you refactor it into a lambda?

Reply:
I don't think I was quite clear enough about what I meant here. I added a more specific comment.